docs/sphinx/sst/running_simeng_with_sst.rst

```diff
@@ -0,0 +1,361 @@
+can be found in the SST component’s documented ports with the ``sst-info`` command.
+
+.. code-block:: python
+
+    component_1 = sst.Component(...) # Assuming component_1 has a port called data_port
+    component_2 = sst.Component(...) # Assuming component_2 has a port called data_port
+
+    link = sst.Link("mylink") # Passing a user-defined link name
+
+    # (component, port_name, port_latency)
+    link.connect((component_1,"data_port","10ns"), (component_2,"data_port","10ns"))
+
+Approach 2
+''''''''''
+For the second approach, the addLink method of each component can be used to establish communication. The addLink method takes in 3 arguments: ``link``, 
+``port_name``, and ``port_latency``.
+
+.. code-block:: python
+
+    component_1 = sst.Component(...) # Assuming component_1 has a port called data_port
+    component_2 = sst.Component(...) # Assuming component_2 has a port called data_port
+
+    link = sst.Link("mylink") # Passing a user-defined link name
+
+    component_1.addLink(link, "data_port", "10ns")
+    component_2.addLink(link, "data_port", "10ns")
+
+Declaring SST SubComponent
+**************************
+If a component has a SubComponent slot, then it can be filled using the ``component.setSubComponent`` method. This method takes in 3 arguments: ``slot_name`` , 
+``element_type`` and ``slot_index``. The ``slot_index`` is the index in which the SubComponent should be inserted. This defaults to 0 and is not required if only 
+one SubComponent is being loaded into the specified slot. Each SubComponent must be loaded into a unique ``slot_index`` and some SubComponents will require the 
+indexes to be incremental.
+
+.. code-block:: python
+
+    import sst
+    # Memory controller from memHierarchy library
+    memctrl = sst.Component("memory", "memHierarchy.MemController")
+    memctrl.addParams({
+        ...
+    })
+    # Memory controller has a subcomponent slot for the memory backend.
+    memory = memctrl.setSubComponent("backend", "memHierarchy.simpleMem")
+    memory.addParams({
+        ...
+    })
+
+SimEng SST Configuration
+------------------------
+
+Configuring the SimEngCore
+**************************
+As discussed earlier, the SimEng core has been wrapped in the ``SimengCoreWrapper`` class and registered as a custom SST Component. Instantiating and using this 
+component is fundamental to running the simulation. The component has the following parameters:
+
+* ``config_path``: Path to YAML configuration file needed by SimEng for configuration of the core microarchitecture under simulation.
+* ``executable_path``: Path to the executable to run inside SimEng.
+* ``executable_args``: Arguments provided to the executable.
+* ``clock``: The frequency of clock ticking the SimEng Core e.g. 1GHz (S.I units accepted).
+* ``max_addr_range``: maximum address which can be accessed by SimEng.
```

Comment:
Should specify if physical or virtual

Reply:
Also; *Maximum...

Reply:
This version of SimEng doesn't deal with the virtual memory system. So currently it's just a memory address because our address space is linear. The notion of physical and virtual addresses comes into play when introducing a virtual memory system. 

I can replace it with "physical address" but then it could lead to a notion of virtual addresses being used somewhere, which could confuse the user, what do you think?